### PR TITLE
[WIP] Feature/link account

### PR DESCRIPTION
**Pull request by SukitOwl**

### CODE REVIEW

File: Cpp/odin-views/link.account.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        auto const credentials = cnx.exec(
+            "SELECT 1 FROM "
+            "odin.identity id "
+            "LEFT JOIN odin.credentials cred "
+            "ON cred.identity_id = id.id "
+            "LEFT JOIN odin.facebook_credentials fb "
+            "ON fb.identity_id = id.id "
+            "LEFT JOIN odin.google_credentials gg "
+            "ON gg.identity_id = id.id "
+            "WHERE id.id='" + identity_id + "' "
```
Comment (tleekkul) [inline Cpp/odin-views/link.account.cpp]: Is this allow the SQL injection? @KayEss Any thought?

Comment (KayEss) [reply Cpp/odin-views/link.account.cpp]: Yes, that's exactly what that is. Let's not do that

Comment (tleekkul) [reply Cpp/odin-views/link.account.cpp]: If not `cnx.exec`, what should we use?

File: Cpp/odin-views/link.account.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                if (not app_user.isnull()) {
+                    throw fostlib::exceptions::not_implemented(
+                            __PRETTY_FUNCTION__,
+                            "The to_account has been linked with app.");
+                }
+
+                if (has_account_registerd(cnx, from_identity_id)) {
+                    throw fostlib::exceptions::not_implemented(
+                            __PRETTY_FUNCTION__,
+                            "The from_account has been registerd.");
```
Comment (tleekkul) [inline Cpp/odin-views/link.account.cpp]: registered

File: Cpp/odin-views/link.account.cpp [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+namespace {
+
+    inline std::pair<boost::shared_ptr<fostlib::mime>, int>
+            respond(fostlib::string message, int code = 403) {
+        fostlib::json ret;
+        if (not message.empty())
+            fostlib::insert(ret, "message", std::move(message));
+        fostlib::mime::mime_headers headers;
+        boost::shared_ptr<fostlib::mime> response(new fostlib::text_body(
+                fostlib::json::unparse(ret, true), headers, "application/json"));
```
Comment (tleekkul) [inline Cpp/odin-views/link.account.cpp]: Should be `html/text` instead.

File: Cpp/odin-views/link.account.cpp (hunk 1 above)
Comment (KayEss) [reply Cpp/odin-views/link.account.cpp]: That's not the problem. Use `%s` in the SQL and then pass a tuple as an extra parameter. Look at the other examples.

Comment (SukitOwl) [reply Cpp/odin-views/link.account.cpp]: Can I use $1 as `WHERE id.id=$1` and
cnx.procedure(sql).exec(std::vector<fostlib::string>{identity_id})

File: Cpp/odin-views/link.account.cpp [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                    __PRETTY_FUNCTION__, "No issuer in the JWT");
+        }
+        auto const jwt_iss = fostlib::coerce<fostlib::string>(jwt_body["iss"]);
+        if (not jwt_iss.startswith(odin::c_app_namespace.value())) {
+            throw fostlib::exceptions::not_implemented(
+                    __PRETTY_FUNCTION__, "App namespace prefix does not match");
+        }
+        auto const app_id =
+                jwt_iss.substr(odin::c_app_namespace.value().code_points());
+        fostlib::json app = odin::app::get_detail(cnx, std::move(app_id));
```
Comment (tleekkul) [inline Cpp/odin-views/link.account.cpp]: Don't move `app_id`

**Pull request merged**